### PR TITLE
feat: add repo logic to coupon, notifs, feedback and profile page

### PR DESCRIPTION
**Pull request by NanoNish**

### CODE REVIEW

File: lib/presentation/feedback/bloc/feedback_page_bloc.dart [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 part 'feedback_page_state.dart';
 
 class FeedbackPageBloc extends Bloc<FeedbackPageEvent, FeedbackPageState> {
-  FeedbackPageBloc()
-      : super(
+  final FeedbackRepository repo;
+  FeedbackPageBloc({
+    required this.repo,
+  }) : super(
           FeedbackPageState(
```
Comment (aman-singh7) [inline lib/presentation/feedback/bloc/feedback_page_bloc.dart:13]: implement new constructor FeedbackPageState.inital() and put the logic in the state.

File: lib/presentation/coupons/bloc/coupons_page_bloc.dart [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 part 'coupons_page_event.dart';
 part 'coupons_page_state.dart';
 
 class CouponsPageBloc extends Bloc<CouponsPageEvent, CouponsPageState> {
-  CouponsPageBloc() : super(const CouponsPageInitialState()) {
+  final CouponRepository repo;
+  CouponsPageBloc({
+    required this.repo,
+  }) : super(const CouponsPageInitialState()) {
     on<CouponsPageFetchEvent>(
```
Comment (aman-singh7) [inline lib/presentation/coupons/bloc/coupons_page_bloc.dart:14]: extract out the logic of on event.

Comment (aman-singh7) [reply lib/presentation/coupons/bloc/coupons_page_bloc.dart:14]: And do this in all the blocs

File: pubspec.lock [hunk 3]
```diff
@@ -173,10 +173,10 @@ packages:
     dependency: transitive
     description:
       name: collection
-      sha256: f092b211a4319e98e5ff58223576de6c2803db36221657b46c82574721240687
+      sha256: "4a07be6cb69c84d677a6c3096fcf960cc3285a8330b4603e0d463d15d9bd934c"
```
Comment (aman-singh7) [inline pubspec.lock:176]: @NanoNish upgrade your flutter to latest version. don't downgrade the dependencies.

**Review by aman-singh7 — CHANGES_REQUESTED** [review]

**Comment by aman-singh7** [comment]
@NanoNish resolve the merge conflicts.

File: pubspec.lock (hunk 3 above)
Comment (NanoNish) [reply pubspec.lock:176]: yessir

**Comment by NanoNish** [comment]
@aman-singh7 resolved the merge conflicts as well

**Review by aman-singh7 — APPROVED** [review]

**Pull request merged**